### ISSUE TITLE
YAMLException: duplicated mapping key at line 315, column 7:

### ISSUE THREAD

**Issue by imbstack** [taskcluster/taskcluster-github#148]
https://sentry.io/taskcluster/taskcluster-github/issues/222528905/

```
YAMLException: duplicated mapping key at line 315, column 7:
          mounts:
          ^
  File "/app/lib/handlers.js", line 241, in Handlers._callee8$
    c = yaml.safeLoad(repoconf);
...
(17 additional frame(s) were not displayed)

YAMLException: duplicated mapping key at line 315, column 7:
          mounts:
          ^
```

**Comment by petemoore** [comment]
Consistently occurs - here is my next push.

Thanks Brian!

https://papertrailapp.com/systems/taskcluster-github/events?focus=770778671832305677&selected=770778671832305677

https://github.com/taskcluster/generic-worker/blob/fb14121f8f16d8aad5da3bd71ebe0c0aebc7fc75/.taskcluster.yml#L315-L316

**Comment by djmitche** [comment]
The error itself is invalid, as far as I can see -- the yaml looks fine.

I think the next debugging step is to pull that file and try running it through the version of the yaml lib we're using, and see if that raises a similar error.

**Comment by petemoore** [comment]
credits to @imbstack  for finding the root cause - it was actually a combination of there being a duplicate elsewhere in the file ([here](https://github.com/taskcluster/generic-worker/blob/fb14121f8f16d8aad5da3bd71ebe0c0aebc7fc75/.taskcluster.yml#L284) and [here](https://github.com/taskcluster/generic-worker/blob/fb14121f8f16d8aad5da3bd71ebe0c0aebc7fc75/.taskcluster.yml#L311)) combined with this [reporting error](https://github.com/nodeca/js-yaml/issues/243).

Thanks guys, and sorry for not spotting the duplication myself!

**Comment by djmitche** [comment]
This is still occurring, and I've verified that #149 is merged to production..

**Comment by petemoore** [comment]
That's strange, I fixed the `.taskcluster.yml` file in generic-worker.

**Comment by imbstack** [comment]
Ah, it's happening in pr-allowed now, instead of in handlers. The nice thing is that the library upgrade worked and is saying that the issue is duplicate artifacts now.

I expect the reason it is failing again in the case of pr-allowed is that the .taskcluster.yml is being loaded from the default branch (probably master?) and it has a duplicate mapping in it.

The fixes for this are twofold:

1. The default branch .taskcluster.yml needs fixed in generic-worker
2. We need to try-catch the yaml loading in pr-allowed.

**Comment by imbstack** [comment]
For bonus points, the message written from pr-allowed should specify that the issue is with .taskcluster.yml on default branch instead of the one on the branch itself.

**Comment by djmitche** [comment]
Also, as @jhford suggested, ideally this doesn't end up reported in sentry..

**Comment by petemoore** [comment]
@djmitche @imbstack with #156 fixed now, can this issue (#148) also be closed? Brian was absolutely right about the master version containing the duplicate.

Let me know if I should intentionally add a duplicate entry again to confirm this issue has really been solved.
